### DISCUSSION TITLE
Instructions for installation with Jupyter Hub from scratch

### DISCUSSION THREAD

**Discussion by joebathelt** [jupyter/nbgrader#1832] [General]
Hello nbgrader community,

I'm grateful for your efforts in developing this valuable tool. I've recently become the instructor for a course that previously employed nbgrader. However, I found that the existing setup was non-functional, prompting me to begin anew. Despite dedicating numerous hours attempting various configurations of Jupyter Hub and nbgrader, I've had little success. 

Could you point me towards a current tutorial detailing the complete setup procedure, including the advised distribution and setup for Jupyter Hub?

Thank you for your help!

Best, 

Joe

**Comment by jeflem** [comment]
Hi @joebathelt,

don't have a tutorial at hand, but have set up a JupyterHub with Nbgrader and LMS (Moodle,...) integration during past months. All config files and install instructions are available online in form of a Podman/Docker image. So you may get some ideas how configuration could look like: [Ananke GitHub repo](https://github.com/jeflem/ananke). Good starting is the file [`images/ananke-nbgrader/Containerfile`](https://github.com/jeflem/ananke/blob/main/images/ananke-nbgrader/Containerfile).

You find my email in the docs. If you have concrete question about your setup don't hesistate to mail me.

Beste regards,

Jens

**Reply by joebathelt** [reply]
Thank you, that looks like a great project. The integration with the LMS is a bit more involved than what I need for this course though.

**Comment by lahwaacz** [comment]
Note that the [Using nbgrader with JupyterHub](https://nbgrader.readthedocs.io/en/stable/configuration/jupyterhub_config.html) documentation page is quite outdated. For example, the format of `c.JupyterHub.load_groups` was changed in JupyterHub 3.2 to

```python
c.JupyterHub.load_groups = {
    'groupname': {
        'users': ['usernames'],
        'properties': {'key': 'value'},
    },
}
```
